src/Dependencies/Collections/RoslynImmutableInterlocked.cs

```diff
@@ -68,7 +68,7 @@ public static bool Update<T>(ref ImmutableSegmentedList<T> location, Func<Immuta
         /// <paramref name="transformer"/> function; otherwise, <see langword="false"/> if the location's value remained
         /// the same because the last invocation of <paramref name="transformer"/> returned the existing value.
         /// </returns>
-        public static bool Update<T, TArg>(ref ImmutableSegmentedList<T> location, Func<ImmutableSegmentedList<T>, TArg, ImmutableSegmentedList<T>> transformer, TArg transformerArgument)
+        public static bool Update<T, TArg>(ref ImmutableSegmentedList<T> location, TArg transformerArgument, Func<ImmutableSegmentedList<T>, TArg, ImmutableSegmentedList<T>> transformer)
```

Comment:
❗ These intentionally mirror [`ImmutableInterlocked`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.immutable.immutableinterlocked?view=net-8.0) and break the drop-in replacement rule.